src/main/java/io/vertx/core/net/TrafficShapingOptions.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2011-2019 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.net;
+
+import io.netty.util.internal.ObjectUtil;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Options describing how {@link io.netty.handler.traffic.GlobalTrafficShapingHandler} will handle traffic shaping.
+ */
+@DataObject(generateConverter = true, publicConverter = false)
```

Comment:
this should be annotated with `io.vertx.codegen.annotations.Unstable` as this might change in the future. I can see several traffic shaping strategies and perhaps we should have the ability to configure also per channel option. We can do this in a follow up PR.